Mage.Sets/src/mage/cards/s/StormOfSouls.java

```diff
@@ -0,0 +1,144 @@
+package mage.cards.s;
+
+import mage.abilities.Ability;
+import mage.abilities.effects.ContinuousEffectImpl;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.ExileSpellEffect;
+import mage.abilities.keyword.FlyingAbility;
+import mage.cards.Card;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.*;
+import mage.filter.StaticFilters;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.players.Player;
+import mage.target.targetpointer.FixedTargets;
+
+import java.util.LinkedHashSet;
+import java.util.Set;
+import java.util.UUID;
+
+/**
+ * @author Alex-Vasile
+ */
+public class StormOfSouls extends CardImpl {
+    public StormOfSouls(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.SORCERY}, "{4}{W}{W}");
+
+        // Return all creature cards from your graveyard to the battlefield.
+        // Each of them is a 1/1 Spirit with flying in addition to its other types.
+        this.getSpellAbility().addEffect(new StormOfSoulsReturnEffect());
+
+        // Exile Storm of Souls.
+        this.getSpellAbility().addEffect(new ExileSpellEffect().concatBy("<br>"));
+    }
+
+    private StormOfSouls(final StormOfSouls card) { super(card); }
+
+    @Override
+    public StormOfSouls copy() { return new StormOfSouls(this); }
+}
+
+class StormOfSoulsReturnEffect extends OneShotEffect {
+    public StormOfSoulsReturnEffect() {
+        super(Outcome.Benefit);
+        staticText = "Return all creature cards from your graveyard to the battlefield." +
+                "Each of them is a 1/1 Spirit with flying in addition to its other types.";
+    }
+
+    private StormOfSoulsReturnEffect(final StormOfSoulsReturnEffect effect) { super(effect); }
+
+    @Override
+    public StormOfSoulsReturnEffect copy() { return new StormOfSoulsReturnEffect(this); }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player player = game.getPlayer(source.getControllerId());
+
+        if (player == null) { return false; }
+
+        Set<Card> creatureCardsToBeMovedFromGraveyard = player.getGraveyard().getCards(StaticFilters.FILTER_CARD_CREATURE, game);
+        if (creatureCardsToBeMovedFromGraveyard == null) { return false; }
+
+        if (creatureCardsToBeMovedFromGraveyard.isEmpty()) { return false; }
+
+        // Returns true if all cards were moved.
+        boolean anyCardsMoved = player.moveCards(creatureCardsToBeMovedFromGraveyard, Zone.BATTLEFIELD, source, game);
+        if (!anyCardsMoved) { return false; }
+
+        // Figure out which cards were successfuly moved so that they can be turned into 1/1 Spirits
+        Set<Card> creatureCardsMovedFromGraveyard = new LinkedHashSet<>();
+
+        for (Card card : creatureCardsToBeMovedFromGraveyard) {
+            if (game.getState().getBattlefield().containsPermanent(card.getId())) {
```

Comment:
There's a comment on this method that indicates it shouldn't be used for this purpose.  Should use check `game.getState().getZone(card.getId) == Zone.BATTLEFIELD` instead.

https://github.com/magefree/mage/blob/9a4bcb6ca5d63170f9c70ea3ef6d0ffb1b4e365d/Mage/src/main/java/mage/game/permanent/Battlefield.java#L166-L176